### PR TITLE
Revm example

### LINKED ISSUE #6145: add example using reth revm sims
### Describe the feature

use reth revm crate for this, very easy to write a example, ask question if you need help on this task 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AbnerZheng**
close https://github.com/paradigmxyz/reth/issues/6145

Add an example of how to use a custom inspector to trace new pending transactions
How to test:
```sh
> cargo run -p custom-inspector -- node --http --ws --dev`
```
and in another contracts project created by `forge`, and then run the following command to deploy a transaction:
```sh
forge create --mnemonic "test test test test test test test test test test test junk" src/Counter.sol:Counter
```

here is the running result from `node`:
```
Inspector result for transaction 0x21abbfee9d8e6af8bce70424425c8d3da92ed63d0d1de89cbd74cef108473e6a: 
 0: PUSH1
2: PUSH1
4: MSTORE
5: CALLVALUE
6: DUP1
7: ISZERO
8: PUSH2
11: JUMPI
16: JUMPDEST
17: POP
18: PUSH1
20: DUP1
21: PUSH2
24: PUSH1
26: CODECOPY
27: PUSH1
29: RETURN
```

### CODE REVIEW

File: examples/custom-inspector/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+struct RethCliTxpoolExt {
+    /// recipients addresses that we want to trace
+    #[arg(long, value_delimiter = ',')]
+    pub recipients: Vec<Address>,
+}
+
+#[derive(Default, Debug, Clone)]
+struct DummyInspector {
+    ret_val: Vec<String>,
+}
```
Comment (mattsse) [inline examples/custom-inspector/src/main.rs:57]: we should add a smol doc here what this inspector does, listing opcodes at the program counter

**Review by mattsse — CHANGES_REQUESTED** [review]
this is great!

only doc nit, and perhaps a few more docs about the what's going on, like
convert the pool transaction
configure the evm with the custom inspector
execute the transaction on a blocking task and await the inspector result

File: examples/custom-inspector/src/main.rs (hunk 1 above)
Comment (AbnerZheng) [reply examples/custom-inspector/src/main.rs:57]: Sure, will do in tomorrow.

File: examples/custom-inspector/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        inspector_handle_register, interpreter::Interpreter, revm, Database, EvmContext, Inspector,
+    },
+    rpc::{
+        compat::transaction::transaction_to_call_request,
+        eth::{revm_utils::EvmOverrides, EthTransactions},
+    },
+    tasks::TaskSpawner,
+    transaction_pool::TransactionPool,
+};
+use reth_revm::interpreter::OpCode;
```
Comment (gakonst) [inline examples/custom-inspector/src/main.rs:31]: we should make this way less verbose. Prelude or otherwise?

Comment (mattsse) [reply examples/custom-inspector/src/main.rs:31]: these are already the reexports -.-

but can def make this nicer

File: examples/custom-inspector/src/main.rs (hunk 1 above)
Comment (AbnerZheng) [reply examples/custom-inspector/src/main.rs:58]: Done.

**Review by mattsse — APPROVED** [review]
nice work

**Pull request merged**